### DISCUSSION TITLE
posts/Go/slice_tricks/

### DISCUSSION THREAD

**Discussion by Q1mi** [Q1mi/BlogComments#324] [Announcements]
https://www.liwenzhou.com/posts/Go/slice_tricks/ 

 李文周 博客 Blog Go语言教程 go 教程 切片 技巧 slice copy delete cut append insert 插入 追加

**Comment by 1974410167** [comment]
感觉可以写一个切片库，实现类似python的remove insert pop 等函数，go这切片操作舒适是丑

**Comment by orivee** [comment]
第二道练习题，我是这样做的：

```go
package main

import (
	"fmt"
	"sort"
)

func main() {
	var a = [...]int{3, 7, 8, 9, 1}
	aSlice := a[0:]

	fmt.Println(a)
	sort.Slice(aSlice, func(i, j int) bool {
		return aSlice[i] < aSlice[j]
	})
	fmt.Println(aSlice)
}
```

**Comment by Johnnylov** [comment]
package main
import ("fmt"
	"sort"
)
func main() {
	var a = [...]int{3, 7, 8, 9, 1}
	sort.Ints(a[:])
	fmt.Println(a)
}

**Comment by seabra2** [comment]
第一道题，是由于初始化了，所以前五个是一个空字符，经过一轮扩容，后面是0-9

**Comment by seabra2** [comment]
内置sort包不能直接对数组进行操作，需要讲数组转为切片使用
func main() {
	var a = [...]int{3, 7, 8, 9, 1}
	sort.Ints(a[:])
	fmt.Println(a) // [1 3 7 8 9]
}
